Final Torch Setup

Raptor1628 opened the pull request:
The last series of changes and updates required before more serious
testing.
Adjusts a lot of access for lockers.
Fixes some broken sprites.
Adds Security Guard gear.
Adds more miltiary haircuts.
Adds some final uniform stuff.
Map fixes and tweaks.
General fixes.
Moves a few more lockers to map datums.
Enables Merchant role.
Inventory changes related to loadout. (That includes adding slot_holster
to the RCD)
Adds custom loadout torch entries. Not sure a better way to do this
other than moving it all to map datums, and considering it's a client
thing, I wasn't sure about doing that. Snapshot said something was in the works, so there's that.

<!-- 
Do not forget to add a changelog when you have made admin/player facing changes that can alter gameplay.
Examples which require a changelog entry include:
* Adding/removing objects that players may interact with, or the way they function.
* Adding/removing/altering admin tools.
* Changing the map.

Examples were changelog entries are optional/not typically required:
* Cosmetic changes such as descriptions, sound effects, etc.
* Optimizations and other changes to underlying systems which do not affect gameplay.
* Minor bug fixes.

You find a README and example file in .\html\changelogs\ for further instructions.
-->

Raptor1628 committed:
Final Torch Setup

The last series of changes and updates required before more serious
testing.
Adjusts a lot of access for lockers.
Fixes some broken sprites.
Adds Security Guard gear.
Adds more miltiary haircuts.
Adds some final uniform stuff.
Map fixes and tweaks.
General fixes.
Moves a few more lockers to map datums.
Enables Merchant role.
Inventory changes related to loadout. (That includes adding slot_holster
to the RCD)
Adds custom loadout torch entries. Not sure a better way to do this
other than moving it all to map datums, and considering it's a client
thing, I wasn't sure about doing that.

NullSnapshot commented:
Odds are that the custom loadout stuff will become embedded into the maps folders. I'd recommend putting it there instead and setting the files up based off of categories EG maps/torch/loadout/accessories or maps/torch/loadout/utility.
Though I'ms till not entirely sure how @PsiOmegaDelta is planning on doing this so it'd be nice to have him chime in.

Raptor1628 commented:
And in true torch/raptor fashion travis failed. More fixes coming right up.

Raptor1628 commented:
Should I also move the exodus ones, or leave them alone?

Raptor1628 committed:
Fixes, moves loadout to map file.

ZCaliber commented:
Don't get my hopes up with that pull name. ;_;

NullSnapshot commented:
UNIT_TEST: /datum/gear/uniform/dress - /obj/item/clothing/under/dress: Did not find the icon state '' in the icon 'icons/obj/clothing/uniforms.dmi'.

NullSnapshot commented:
Dont touch exodus loadouts yet because if you do that it'll break the current loadout system. There's no code for different loadout tables yet.

thasc commented on code/modules/client/preference_setup/loadout/torch_loadout_accessories.dm:
```diff
@@ -0,0 +1,489 @@
+	display_name = "suspenders"
+	path = /obj/item/clothing/accessory/suspenders
+	allowed_roles = list("Research Director", "NanoTrasen Liaison", "Scientist", "Prospector", "Research Assistant",
+						"SolGov Representative", "Passenger", "Maintenance Assistant", "Roboticist""Supply Assistant", "Bartender", "Merchant")
```
Travis failure due to missing a comma between "Roboticist" and "Supply Assistant".

Kasuobes commented on maps/torch/loadout/loadout_uniform.dm:
```diff
@@ -0,0 +1,424 @@
+	path = /obj/item/clothing/under/dress
+	allowed_roles = list("Scientist", "Prospector", "Security Guard", "Research Assistant",
+						"Passenger", "Maintenance Assistant", "Roboticist", "Medical Assistant",
+						"Virologist", "Chemist", "Counselor", "Supply Assistant", "Bartender", "Merchant)
```
Missing " after merchant.

Kasuobes commented on maps/torch/loadout/loadout_head.dm:
```diff
@@ -0,0 +1,328 @@
+						"Senior Engineer", "Engineer", "Brig Officer", "Forensic Technician", "Master at Arms", "Senior Physician", "Physician", "Deck Officer",
+						"Deck Technician", "Sanitation Technician", "Cook", "Crewman")
+
+/datum/gear/head/fleetutility
```
Caps and covers cost points while other uniform items don't.

comma commented on code/game/machinery/vending.dm:
```diff
@@ -800,7 +802,7 @@
 	icon_deny = "med-deny"
 	product_ads = "Go save some lives!;The best stuff for your medbay.;Only the finest tools.;Natural chemicals!;This stuff saves lives.;Don't you want some?;Ping!"
-	req_access = list(access_medical_equip)
+	req_access = list(access_medical)
```
That'd let janitors access it.

Raptor1628 committed:
Fixes and Updates

Raptor1628 committed:
Map Tweaks

Raptor1628 committed:
More fixes and tweaks.

Raptor1628 committed:
Even more fixes

Raptor1628 committed:
Adds a dress icon_state

Kasuobes commented on code/modules/client/preference_setup/loadout/torch_loadout_uniform.dm:
```diff
@@ -0,0 +1,424 @@
+	path = /obj/item/clothing/under/dress
+	allowed_roles = list("Scientist", "Prospector", "Security Guard", "Research Assistant",
+						"Passenger", "Maintenance Assistant", "Roboticist", "Medical Assistant",
+						"Virologist", "Chemist", "Counselor", "Supply Assistant", "Bartender", "Merchant)
```
" after merchant.

Raptor1628 committed:
More fixes, attempt at placating travis.

Raptor1628 committed:
Shuffles sprites. Fixes

Raptor1628 committed:
More wrestling with travis.

Raptor1628 committed:
Fixes

Minijar commented:
You'll get it eventually. *cough travis is a bitch cough*

Raptor1628 commented:
Actually it's just like 4-5 people being completely blind.

thasc commented:
huzzah

comma reviewed (approved):
Could use a squashing tho.

PsiOmegaDelta commented on code/modules/mob/new_player/sprite_accessories.dm:
```diff
@@ -193,9 +193,11 @@
 	fringetail
 		name = "Fringetail"
 		icon_state = "hair_fringetail"
+*/
```
So remove it?

PsiOmegaDelta commented on code/modules/mob/new_player/sprite_accessories.dm:
```diff
@@ -193,9 +193,11 @@
 		icon_state = "hair_ponytail5"
 		gender = FEMALE
 
+/*Apparently this doesn't exist
```
So remove it?

PsiOmegaDelta commented:
What's the ```MapStaging/uniforms.dmi``` file all about?

PsiOmegaDelta commented on maps/torch/structures/closets/command.dm:
```diff
@@ -0,0 +1,98 @@
+	)
+
+/obj/structure/closet/secure_closet/sea/New()
+	..()
```
Why these New() overrides with no effects at all?

Raptor1628 replied:
A few of the closets that you had transferred over to torch files had that on them, like Liaison in misc.dm, so I wasn't sure if it was needed or not.

PsiOmegaDelta replied:
I messed up when transferring things to will_contain then, unless they had code that did more than just call ..()

Kasuobes commented on maps/torch/items/machinery.dm:
```diff
@@ -0,0 +1,16 @@
+	icon_deny = "med-deny"
+	product_ads = "Go save some lives!;The best stuff for your medbay.;Only the finest tools.;Natural chemicals!;This stuff saves lives.;Don't you want some?;Ping!"
+	req_access = list(access_medical)
+	products = list(/obj/item/weapon/reagent_containers/glass/bottle/antitoxin = 4,/obj/item/weapon/reagent_containers/glass/bottle/inaprovaline = 4,
```
Subtype of the standard thing, you can leave out everything else except the changed access if you haven't changed anything else.

Finalsong commented:
The random items that spawn in things like lockers, cabinets, etc- Probably the solgov type ones, all spawn as "random maintenance (type) item" https://gyazo.com/5e6f4c7d1c0d5ffaab2f31b8abffcbf2

Finalsong commented:
Random items in lockers spawn as "random (item type) item" https://gyazo.com/5e6f4c7d1c0d5ffaab2f31b8abffcbf2

Raptor1628 committed:
Fixes

comma committed:
Adds support for per-branch outfits for jobs

Converts XO job/outfits to it for example.

Raptor1628 commented:
Chinsky is adding his branch-based outfit stuff, so waiting on that.

comma committed:
Added support for autoequip rank accessories

If rank datum has type defined, it will stick one onto the guy. Again, only done one for example (Marine Major)
Rearranged copy_to a bit since both branch and rank are needed for proper equipping.

comma commented:
Okay can now define per branch outfit variations, and ranks can spawn their own bling if it's defined.
Only done one of each system for example, tons of copy/paste work needed to convert rest.
But that way you won't have to rely on players picking right thing from loadout.

Raptor1628 commented:
Alright, if travis passes on this, could this PR be merged, then I'll do all the copy-paste setup, and preferably tick torch.dm, in another PR.
